bears/general/AnnotationBear.py

```diff
@@ -28,104 +27,212 @@ def run(self, filename, file, language: str, coalang_dir: str = None):
-    :param inside_range:   A tuple SourceRange objects identifying ranges
-                           to be searched in.
-    :return:               True if inside_range is found inside any of the
-                           ranges given by outside_ranges, else
-                           False is returned.
-    """
-    for outside_range in outside_ranges:
-        if inside_range == outside_range:
-            continue
-        # Special case of python language.
-        # Where doc strings (""") and strings (") have a similar start.
-        elif inside_range.start == outside_range.start:
-            if inside_range.end < outside_range.end:
-                return True
-        elif (inside_range.start > outside_range.start and
-              inside_range.start <= outside_range.end):
-            return True
-    return False
+        end_start = text.find(annotation_end,
+                              position + 1)
+        end_end = end_start + len(annotation_end) - 1
+        if end_start == -1:
+            _range = SourceRange.from_absolute_position(
+                filename, AbsolutePosition(file, position))
+            raise NoCloseError(annotation_start, _range)
+
+        return SourceRange.from_absolute_position(
+            filename,
+            AbsolutePosition(file, position),
+            AbsolutePosition(file, end_end)), end_end
+
+    @staticmethod
+    def get_singleline_strings(file, filename,
+                               text,
+                               string_start, string_end,
+                               position):
+        """
+        Get sourcerange of a single-line string and its end position.
+
+        :param file:
+            A tuple of strings, with each string being a line in the file.
+        :param filename:
+            The name of the file.
+        :param string_start:
+            The string which specifies how a string starts.
+        :param string_end:
+            The string which specifies how a string ends.
+        :position:
+            An integer identifying the position where the string started.
+        :return:
+            A SourceRange object identifying the range of the single-line
+            string and the end_position of the string as an integer.
+        """
+        end_position = text.find(string_end,
+                                 position + 1) + len(string_end) - 1
+        newline = text.find("\n", position + 1)
+        if newline == -1:
+            newline = len(text)
+        if end_position == -1:
+            _range = SourceRange.from_absolute_position(
+                filename, AbsolutePosition(file, position))
+            raise NoCloseError(string_start, _range)
+        if newline > end_position:
+            return SourceRange.from_absolute_position(
+                filename,
+                AbsolutePosition(file, position),
+                AbsolutePosition(file, end_position)), end_position
```

Comment:
the last param isn't inside the paranthesis hence python has a problem with it being in a newline unless i escape it, i'm leaving it as is for now, will change if necessary